File: sol/3133/3133.py
class Solution:
    def minEnd(self, n: int, x: int) -> int:
        m = 1
        n -= 1
        res = x
        while n != 0:
            if (res & m) == 0:
                if n & 1:
                    res |= m
                n >>= 1
            m <<= 1
        return res

File: sol/3133/3133.cpp
struct Solution {
    using ll = long long;
    long long minEnd(int n, int x) {
        ll m = 1;
        --n;
        ll res = x;
        while (n != 0){
            if ((res & m) == 0){
                if (n & 1){
                    res |= m;
                }
                n >>= 1;
            }
            m <<= 1;
        }
        return res;
    }
};
